StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/ReadabilityRules/SA1129CodeFixProvider.cs

```diff
@@ -58,12 +58,32 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
         private static SyntaxNode GetReplacementNode(SyntaxNode node)
         {
             var newExpression = (ObjectCreationExpressionSyntax)node;
+            var identifierName = (newExpression.Type as IdentifierNameSyntax)
+                ?.Identifier.Text;
 
-            return SyntaxFactory.DefaultExpression(newExpression.Type)
+            SyntaxNode replacement = null;
+            if (identifierName.Equals(nameof(CancellationToken), System.StringComparison.OrdinalIgnoreCase))
+            {
+                replacement = GetCancellationTokenNoneSyntax();
+            }
+            else
+            {
+                replacement = SyntaxFactory.DefaultExpression(newExpression.Type);
+            }
+
+            return replacement
                 .WithLeadingTrivia(newExpression.GetLeadingTrivia())
                 .WithTrailingTrivia(newExpression.GetTrailingTrivia());
         }
 
+        private static SyntaxNode GetCancellationTokenNoneSyntax()
+        {
+            return SyntaxFactory.MemberAccessExpression(
```

Comment:
This needs to start as fully-qualified and then be simplified (using `Simplifier`) rather than assuming `CancellationToken` is in scope.